presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java

```diff
@@ -444,6 +445,16 @@ public static boolean isS3FileSystem(String user, HdfsEnvironment hdfsEnvironmen
         }
     }
 
+    public static boolean isViewFileSystem(String user, HdfsEnvironment hdfsEnvironment, Path path)
+    {
+        try {
+            return hdfsEnvironment.getFileSystem(user, path) instanceof ViewFileSystem;
```

Comment:
This will break for Hadoop 1.x which doesn't have `ViewFileSystem`. Unfortunately, we'll need to compare the name against a hard-coded string `org.apache.hadoop.fs.viewfs.ViewFileSystem`.